### ISSUE TITLE
SerializerGenerator does not supports IList generation when IsRecursive = true

### ISSUE THREAD

**Issue by neuecc** [msgpack/msgpack-cli#203]
When invoke GenerateSerializerSourceCodes with `IList<>`, usually it works.
But if  ` SerializerCodeGenerationConfiguration.IsRecrusive = true`, throws excpetion.

repro code.

```csharp
// Ok.
SerializerGenerator.GenerateSerializerSourceCodes(new[] { typeof(IList<int>) });

// Ng.
SerializerGenerator.GenerateSerializerSourceCodes(new SerializerCodeGenerationConfiguration() { IsRecursive = true }, new[] { typeof(IList<int>) });
```

exception message.

```
System.Runtime.Serialization.SerializationException: Cannot serialize type 'System.Collections.Generic.IList`1[System.Int32]' because it does not have any serializable fields nor
```

**Comment by yfakariya** [comment]
Thank you for reporting. I will investigate and fix this.

**Comment by yfakariya** [comment]
Fixed in 20ae0c3
